pages/[id]/minigames/merge-pets/index.js

```diff
@@ -0,0 +1,430 @@
+    for (let i = 0; i < 4; i++) {
+      for (let j = 4 - 2; j >= 0; j--) {
+        if (newGrid[j][i] !== 0) {
+          let k = j;
+          while (k < 4 - 1 && newGrid[k + 1][i] === 0) {
+            newGrid[k + 1][i] = newGrid[k][i];
+            newGrid[k][i] = 0;
+            k++;
+          }
+          if (k < 4 - 1 && newGrid[k + 1][i] === newGrid[k][i]) {
+            newGrid[k + 1][i] *= 2;
+            newGrid[k][i] = 0;
+          }
+        }
+      }
+    }
+    setGrid(newGrid);
+    newNumber();
+  }
+
+  function newNumber() {
+    const emptyCells = [];
+
+    // Run through the grid to find empty cells
+    for (let j = 0; j < 4; j++) {
+      for (let i = 0; i < 4; i++) {
+        if (grid[j][i] === 0) {
+          emptyCells.push([j, i]);
+        }
+      }
+    }
+
+    // Check if there are empty cells
+    if (emptyCells.length === 0) {
+      gameOver();
+      return;
+    }
+
+    // Random selection of an empty cell
+    const randomIndex = Math.floor(Math.random() * emptyCells.length);
+    const [newX, newY] = emptyCells[randomIndex];
+
+    // Set the new number in the random cell
+    const newGrid = [...grid];
+    newGrid[newX][newY] = 2;
+    setGrid(newGrid);
+  }
+
+  function getPoints() {
+    let sum = 0;
+    grid.forEach((row) => {
+      row.forEach((point) => (sum += point));
+    });
+    return sum;
+  }
+
+  function gameOver() {
+    const points = getPoints();
+    const money = Math.floor(points / 8);
+    setConfirmationPopUpContent({
+      ...confirmationPopUpContent,
+      show: true,
+      message: `Game over, your high score is: ${points}. For this you get ${money} coins!`,
+      onConfirm: () => {
+        onAddMoney(money);
+        router.push(`/pet-detail-page/${id}`);
+      },
+    });
+  }
+
+  function renderSquareContent(value) {
+    if (value !== 0) {
+      return animalImage ? (
+        <ImageContainer>
+          <Image
+            src={petImageMap[value]}
+            alt="Pet Image"
+            width={40}
+            height={40}
+          />
+        </ImageContainer>
+      ) : (
+        value
+      );
+    }
+    return "";
+  }
+
+  function renderSquareLegend() {
+    let content = [];
+
+    Object.entries(petImageMap).forEach(([val, imgUrl], index) => {
+      const backgroundColor = `zahl${val}`;
+      content.push(
+        <Square key={index} $squareColor={backgroundColor}>
+          <Image src={imgUrl} alt="Pet Image" width={40} height={40} />
+        </Square>
+      );
+      // Add arrow if not the last item
+      if (index !== Object.keys(petImageMap).length - 1) {
+        content.push(<span key={`arrow${index}`}>&rarr;</span>);
+      }
+    });
+
+    return <StyledLegend>{content}</StyledLegend>;
+  }
+
+  const toggleDisplay = () => {
+    setAnimalImage(!animalImage);
+  };
+
+  return (
+    <>
+      <header>
+        <StyledLeftButton onClick={() => router.push(`/pet-detail-page/${id}`)}>
+          Back
+        </StyledLeftButton>
```

Comment:
As mentioned before - it is better to use the Next.js `<Link> `component for navigation.